File: perl/Day1/part1.pl
#!/usr/bin/perl

#open(FH, '<', "sample-input.txt") or die $!;
open(FH, '<', "full-input.txt") or die $!;

my $doubleDigitRe = qr/^[^\d]*(\d).*(\d)[^\d]*$/;
my $singleDigitRe = qr/^[^\d]*(\d)[^\d]*$/;

$accum = 0;
while (<FH>) {
    if ($_ =~ /${doubleDigitRe}/) {
        $accum += ($1.$2);
    } elsif (/${singleDigitRe}/) {
        $accum += ($1.$1);
    }
}

print $accum . "\n";

close(FH);

File: python/Day1/part1.py
from functools import reduce

def just_digits(str):
    return ''.join(c for c in str if c.isdigit())

def first_last(str):
    return str[0] + str[-1]

def str_add(a, b):
    return int(a) + int(b)

#f = open("sample-input.txt", "r")
f = open("full-input.txt", "r")
print(reduce(str_add, map(first_last,map(just_digits, map(str.rstrip, f.readlines())))))

f.close()